fdbclient/include/fdbclient/BlobRestoreCommon.h

```diff
@@ -35,4 +35,44 @@ struct BlobGranuleBackupConfig : public KeyBackedClass {
 	KeyBackedProperty<std::string> mutationLogsUrl() { return subspace.pack(__FUNCTION__sr); }
 };
 
-#endif
\ No newline at end of file
+// Defines blob restore state
+enum BlobRestorePhase {
+	UNINIT = 0,
+	INIT = 1,
+	STARTING_MIGRATOR = 2,
+	LOADING_MANIFEST = 3,
+	LOADED_MANIFEST = 4,
+	COPYING_DATA = 5,
+	APPLYING_MLOGS = 6,
+	DONE = 7,
+	ERROR = 8,
+	MAX = 9
+};
+
+template <>
+inline Standalone<StringRef> TupleCodec<BlobRestorePhase>::pack(BlobRestorePhase const& val) {
+	return Tuple::makeTuple(static_cast<int>(val)).pack();
+}
+template <>
+inline BlobRestorePhase TupleCodec<BlobRestorePhase>::unpack(Standalone<StringRef> const& val) {
```

Comment:
I think we could likely do this with a templated `TupleCodec` that works for all enums. It would need to use something like `std::enable_if_t<std::is_enum_v<Enum>>>`.